Sources/xcodeproj/Objects/Files/PBXContainerItemProxy.swift

```diff
@@ -8,10 +8,10 @@ public final class PBXContainerItemProxy: PBXObject {
         case other
     }
 
-    /// The object is a reference to a PBXProject element.
+    /// The object is a reference to a PBXProject element if proxy is for the object located in current .xcodeproj, otherwise PBXFileReference.
     var containerPortalReference: PBXObjectReference
 
-    /// Returns the project that contains the remote object.
+    /// Returns the project that contains the remote object. If container portal is a remote project this getter will fail. Use isContainerPortalFileReference to check if you can use the getter
```

Comment:
Considering that we are in the middle of a major release, I'd take the opportunity to make the API nicer. What about something like:

```swift
enum ContainerPortal {
  case project(PBXProject)
  case fileReference(PBXFileReference)
}

// PBXContainerItemProxy
class PBXContainerItemProxy {
  public var containerPortal: ContainerPortal
}
```

What do you think @damirdavletov ?

Reply:
@pepibumur I agree regarding making the API nicer. My objective was not introduce breaking changes, but if that's not problem I'll try to make it nicer. Thanks for the feedback!

Reply:
Yeah, we avoid introducing breaking changes but this is a special case because we are working on a major release. Feel free to leverage Swift features to make the API nice :)